### PR TITLE
Content Helper: Fix top referrers percentage displaying as "NaN"

### PR DESCRIPTION
**Pull request by acicovic**
## Description
In certain cases, the top referrers percentage in the Content Helper would display as `NaN`. This was not a problem in the JS side, but rather an erroneous calculation in PHP which resulted in this.

## Motivation and context
Fix bugs!

## How has this been tested?
1. We first introduced integration tests that expose the issue.
2. We committed the fix that solves the issue.
3. The tests passed.

### CODE REVIEW

**Comment by acicovic** [comment]
Marking this as ready for review, though SonarCloud complains about duplication in a test file.

File: src/Endpoints/class-referrers-post-detail-api-proxy.php [hunk 1]
```diff
@@ -212,6 +216,17 @@ private function generate_referrers_data(
...
+	/**
+	 * Converts a string to a positive integer, removing any non-numeric
+	 * characters.
+	 *
+	 * @param string $string The string to be converted to an integer.
+	 * @return int The integer resulting from the conversion.
+	 */
+	private function convert_to_positive_integer( string $string ): int {
+		return (int) preg_replace( '/\D/', '', $string );
+	}
```
Comment (mehmoodak) [inline src/Endpoints/class-referrers-post-detail-api-proxy.php]: Should we move this to `utils.php`?

**Review by mehmoodak — APPROVED** [review]
Left a minor comment but approving the PR.

File: src/Endpoints/class-referrers-post-detail-api-proxy.php (hunk 1 above)
Comment (acicovic) [reply src/Endpoints/class-referrers-post-detail-api-proxy.php]: Good catch!

**Pull request merged**